src/test/resources/application-test.yml

```diff
@@ -1,6 +1,7 @@
 spring:
   main:
     banner-mode: 'off'
+    allow-bean-definition-overriding: true
```

Comment:
So the problem here is that it's picking up `RestClient` in other tests which have the same name (https://github.com/polysantiago/spring-boot-rest-client/blob/master/src/main/java/io/github/polysantiago/spring/rest/RestClientsRegistrar.java#L150). Changing this to a more generic/specific name would mean a few changes (https://github.com/polysantiago/spring-boot-rest-client/blob/master/src/main/java/io/github/polysantiago/spring/rest/RestClientsRegistrar.java#L165 and `RestClientContext`). Using this property fixes it but I'm wondering if we shouldn't fix the name conflict...? Could this actually happen in production?

Reply:
Ah, right, now I realise what the problem is! I think it's fair to assume that the RestClients have a unique name, cause it's also tied to the configuration of the resources. Feels fair that the users of the library should follow that rule. 

I can change the test code a bit to use unique names for the clients. Or what do you think?

Reply:
Is there any way we could make the test cases isolated from each other? Ideally, one test should not pick the configuration from another.

Reply:
Ok, there's an easier solution.
Changing `basePackageClasses` attribute to `clients` on `@EnableRestClients` both on `RestClientHateoasTest` and `RestClientTest` fixes the problem. However, these lines https://github.com/polysantiago/spring-boot-rest-client/blob/master/src/main/java/io/github/polysantiago/spring/rest/RestClientsRegistrar.java#L67-L74 are no longer covered by tests. Guess we can create a unit test for that. What do you think?

Reply:
Mmmm, what are you suggesting exactly? I changed at least the RestClients so that they should not share the application config anymore, but is there something else they still share?

Reply:
The problem is that we're scanning the whole package. Nevermind. I'll merge it like this 👍